### ISSUE TITLE
MenuContext positioning means mapping does not work 

### ISSUE THREAD

**Issue by alfie-ab** [instea/react-native-popup-menu#28]
First of all, thanks for your library it's been very useful!

So, I understand why MenuContext needs to be at the top of the component hierarchy but in my app I'm mapping cards, like so:

![screen shot 2017-04-11 at 12 32 49](https://cloud.githubusercontent.com/assets/20194492/24907618/c9c0f780-1eb4-11e7-8bf2-f4de6f5d1a1f.png)

Code mapping:
```
  return filteredSortedArray.map(post =>
        <ExamCardDetail
          key={post["Date, time"]}
          dateTime={post["Date, time"]}
          type={post["type"]}
          body={post["body"]}
          keyId={post["key"]}
          caseId={post["caseId"]}
        />
      )
```

The component ExamCardDetail contains the rest of the popupmenu code, like so:

```
        <Menu>
              <MenuTrigger>
                <Icon
                  name="ios-arrow-down"
                  size={25}
                  color="grey"
                />
              </MenuTrigger>
              <MenuOptions>
                <MenuOption style={optionsStyle}>
                  <Icon.Button
                    name="md-create"
                    size={25}
                    onPress={() => this._editExam()}
                    color="red"
                    backgroundColor="white"
                  />
                  <Text style={textStyle}>Edit</Text>
                </MenuOption>
                <MenuOption style={optionsStyle}>
                  <Icon.Button
                    name="ios-trash"
                    size={25}
                    onPress={() => this._deleteExam()}
                    color="red"
                    backgroundColor="white"
                  />
                  <Text style={textStyle}>Delete</Text>
                </MenuOption>
              </MenuOptions>
           </Menu>
```

My popup menu on each of these cards contains the delete function, and so I pass down the unique id of the card so it can be deleted from Firebase. However because I am not mapping MenuContext (as it needs to be at the top of the hierarchy) it is unable to receive the ids of the different cards. It only receives the last one that was mapped. Therefore when I try to delete a specific card it is only able to delete the latest one.

If I try to map MenuContext as a wrapper around the card, this happens:

![screen shot 2017-04-11 at 12 15 21](https://cloud.githubusercontent.com/assets/20194492/24907272/60bfeee0-1eb3-11e7-9e5e-9f07f6a97b01.png)

Code mapping:

```
return filteredSortedArray.map(post =>
        <MenuContext>
          <ExamCardDetail
            key={post["Date, time"]}
            dateTime={post["Date, time"]}
            type={post["type"]}
            body={post["body"]}
            keyId={post["key"]}
            caseId={post["caseId"]}
          />
        </MenuContext>
      )
```

This works, but obviously looks very bad. I have tried to style MenuContext, Menu and the other Components to fix this but it does not work.  

Basically my question is: is there a workaround for this? Is there a way to map MenuContext (i.e. replace it's high hierarchical position at the top of my app) and still maintain it's default positioning and styling?

Sorry for such a longwinded issue! Any help would be much appreciated

**Comment by sodik82** [comment]
Hi, thanks for the explanation.

If I understand correctly, your main problem is how to delete specific CardDetail. The working example for that can be found here https://github.com/instea/react-native-popup-menu/blob/master/examples/Example.js#L105

The main idea is to give information about the card to your delete handler. Something like
```
onPress={() => this._deleteExam(this.props.key)}
```

Additionally I would recommend to use directly `onSelect` property of menu option (if you don't have special reasons - it is not mandatory) and just use simple icon inside. e.g. 
```
<MenuOption onSelect={() => this._deleteExam(this.props.key)}>
           <Icon
              name="ios-trash"
              size={25}
              color="red"
              backgroundColor="white"
            />
            <Text style={textStyle}>Delete</Text>
</MenuOption>
```
Hope it helped, let me know if you get it working :)

**Comment by alfie-ab** [comment]
Hi, thanks for the quick response.

Sorry I should have included more of my code, this is my delete function:

```
_deleteExam = () => {
    firebaseHelper._deleteData(props.caseId, "examinations", props.keyId)
  }
```

Which connects to this one in a firebaseHelpers file:

```
module.exports._deleteData = (caseId, examOrInv, keyId) => {
	firestack.database().ref("cases").child(caseId).child(examOrInv).child(keyId).remove()
}
```

So I pass the correct and specific props for that card, the issue is that when MenuContext is not directly mapped - as in the first example in my original comment - it (MenuContext) is only rendered once and so only receives the last mapping of ExamCardDetail and thus only one props rather than two.

In my code MenuContext is rendered one or maybe even two level(s) higher than in the example code you posted. I have the screen itself, then the specific tab, then the feed underneath that tab (where I map the cards) and in order for it to style properly I need to put MenuContext in the component that is the screen itself. Any lower down and it renders like in the second screen shot.

Basically, is there a way to map bring MenuContext down a few levels of the hierarchy and still maintain it's styling so that it does not render as it does in my second screenshot in the original comment?

Hope that's clearer!

**Comment by alfie-ab** [comment]
I can post more of my code if needed, it's just it goes over four files so I don't know how relevant it would be!

**Comment by sodik82** [comment]
I believe there is a misconception what `MenuContext` is about. MenuContext e.g. defines where menu backdrop is. It has nothing with any kind of context for your menu. Thus you can have multiple contexts but e.g. closing of menu by clicking outside might not work if you click into another menu context. 

Thus I suggest to change your  `_deleteExam` method to
```
_deleteExam = (caseId, keyId) => {
    firebaseHelper._deleteData(caseId, "examinations", keyId)
  }
```
and then call it as suggested above 
```
onSelect={() => this._deleteExam(this.props.caseId, this.props.keyId)}
```
This way "menu" press handler will be properly bound!

If you realy have to use multiple context, you can change its styling as described in https://github.com/instea/react-native-popup-menu/blob/master/doc/api.md#custom-styles (see 2nd note). But it is not officially recommended!

**Comment by alfie-ab** [comment]
You're right, it works! Thanks a lot for your help - I should have read the docs closer! I'll close the issue
